dev/com.ibm.ws.webcontainer.servlet.3.1/src/com/ibm/ws/webcontainer31/srt/SRTServletRequest31.java

```diff
@@ -121,7 +120,6 @@ public void initForNextRequest(IRequest req) {
 
             this._request = req;
             _srtRequestHelper = getRequestHelper();
-            SRTServletRequestThreadData.getInstance().init(null);
```

Comment:
I think this is the main point of this function.  This should call srtRequestDataInitNull() I believe.

Reply:
Hmmm, hadn't noticed before but SRTServletRequest.initForNextRequest(req) is identical to SRTServletRequest31.initForNextRequest(req) except for getContentLength() vs getContentLengthLong() and resetting the upgrade status ...
`this.upgradeCalledAndSuccessful = false;`
Seems dumb to have all that duplicate code, but not sure if it is worth refactoring around it ...